src/components/Game.js

```diff
@@ -58,31 +59,44 @@ class Game extends Component {
             onTileClick={this.onTileClick}
           />
         </div>
-        <GameStats seconds={this.state.seconds} moves={this.state.moves} />
+        <GameStats
+          seconds={this.state.seconds}
+          moves={this.state.moves}
+          gameState={this.state.gameState}
+        />
       </div>
     );
   }
 
-  _isGameOver() {
-    return false;
+  _isGameOver(tiles) {
+    if (
+      tiles.filter(tile => {
+        return tile.tileId + 1 === tile.number;
+      }).length === 16
```

Comment:
You could use `props.gridSize ** 2` to cover the case of a grid that is bigger than 4 by 4.